File: Java/FactoryPattern/DatabaseFactory/src/com/laxmena/db/PostgresDatabase.java
package com.laxmena.db;

public class PostgresDatabase implements Database {
    public PostgresDatabase() {
        System.out.println("============= Postgres Database =============");
    }

    @Override
    public void createTable(String tableName) {
        System.out.println("Creating Table " + tableName + "...");
    }

    @Override
    public void queryTable(String tableName, String query) {
        System.out.println("Querying " + query + " | on Table: " + tableName);
    }
}

File: Python/FactoryPattern/DatabaseAdapter/db/postgres_database.py
from db.database import Database


class PostgresDatabase(Database):
    def __init__(self):
        print("======== Postgres ========")

    def create_table(self, table_name: str) -> None:
        print("PostgreSQL table creation: " + table_name)

    def query_table(self, table_name: str, fields: str) -> None:
        print("Querying Table: " + table_name + " for fields " + fields)
